xc/xc7/fasm2bels/ioi_models.py

```diff
@@ -87,7 +142,183 @@ def process_idelay(top, features):
     top.add_site(site)
 
 
+def process_iserdes(top, site, idelay_site=None):
+    """
+    Decodes the ISERDES primitive
+    """
+
+    # ISERDES
+    bel = Bel('ISERDESE2')
+
+    data_rate = None
+    if site.has_feature("ISERDES.DATA_RATE.SDR"):
+        data_rate = '"SDR"'
+    else:
+        data_rate = '"DDR"'
+    bel.parameters['DATA_RATE'] = data_rate
+
+    # TODO: There shouldn't be mixed width in FASM features.
```

Comment:
This feels like it could crop up in a new fasm2bels failure in the future.  Is it possible to add a failing test now to demostrate the issue you are mentioning here?

Reply:
@litghost You mean add it and then disable? (we want green CI). I can think of a test with data width of eg. 4 that cannot be decoded now.

Reply:
> @litghost You mean add it and then disable? (we want green CI). I can think of a test with data width of eg. 4 that cannot be decoded now.

That is one way to go.  I think the valid options are:
 - Create and issue, and attach a failing test case so that it can be fixed in the future
 - Create the failing test case and fix it in this PR

The TODO by itself is insufficient.

Reply:
I'll go with an issue then. Solving it probably requires adjustment to prjxray fuzzers, and I think that this PR is already kind of big.

Reply:
There: https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1589